bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/FlatRowAdapter.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.scanner;
+
+import com.google.cloud.bigtable.data.v2.models.RowAdapter;
+import com.google.cloud.bigtable.grpc.scanner.FlatRow.Cell;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Adapter for {@link RowAdapter} that uses {@link FlatRow}'s to represent logical rows.
+ */
+public class FlatRowAdapter implements RowAdapter<FlatRow> {
+
+  @Override
+  public RowBuilder<FlatRow> createRowBuilder() {
+    return new FlatRowAdapter.FlatRowBuilder();
+  }
+
+  public class FlatRowBuilder implements RowBuilder<FlatRow> {
+    private ByteString currentKey;
+    private Map<String, List<Cell>> cells = new TreeMap<>();
+    private List<Cell> familyCells = null;
```

Comment:
Please group these with the related fields previousFamily & totalCellCount. Also please add a comment to help the reader disambiguate between them

Reply:
Added [this](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/2034/files#diff-e2e90efec4bfb8d9914f43602303a3dfR47) comment for the map, Changed `familyCells` to `currentFamilyCells` to make it more descriptive.